Ethernet: Remove V1 code for stm32H7 and stm32H5 series

marwaiehm-st opened the pull request:
- Add stm32h7 ethernet compatible "st,stm32h7-ethernet", used also for stm32h5.
- Remove the V1 code for STM32H7 and STM32H5 series
- Replace "CONFIG_SOC_SERIES_STM32H7X || CONFIG_SOC_SERIES_STM32H5X"
by the compatble st_stm32h7_ethernet

marwaiehm-st committed:
drivers: ethernet: eth_stm32_hal: Remove V1 code for H7/H5 series

Remove the V1 code for STM32H7 and STM32H5 series

Signed-off-by: IBEN EL HADJ MESSAOUD Marwa <marwa.ibenelhadjmessaoud-ext@st.com>

marwaiehm-st committed:
dts: Introduce stm32h7 ethernet compatible

Add stm32h7 ethernet compatible "st,stm32h7-ethernet",
used also for stm32h5.

Signed-off-by: IBEN EL HADJ MESSAOUD Marwa <marwa.ibenelhadjmessaoud-ext@st.com>

erwango commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -1486,27 +1296,30 @@ static const struct eth_stm32_hal_dev_cfg eth0_config = {
+					.MediaInterface = IS_ENABLED(CONFIG_ETH_STM32_HAL_MII)
+								  ? ETH_MEDIA_INTERFACE_MII
+								  : ETH_MEDIA_INTERFACE_RMII,
+			},
```
I'm not sure about the indentation shift in this would section.
If really required, please make it in a dedicated commit.

marwaiehm-st replied:
It's required in the compilation checks. I can remove it.
![Screenshot from 2024-10-18 12-01-54](https://github.com/user-attachments/assets/4cd5c9e7-6391-4ad8-9e91-8fc5f65b120b)

erwango replied:
Ok, these are just warnings which won't affect result warning, so not mandatory to fix.
In general case, if you chose to fix them, please do it in a dedicated commit.

erwango commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -1706,27 +1519,27 @@ static int ptp_stm32_init(const struct device *port)
-		&ptp_hclk_rate);
+				     (clock_control_subsys_t)&eth_cfg->pclken_ptp,
+#endif /* DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_ethernet) */
+				     &ptp_hclk_rate);
```
Please don't mix indentation changes with the clean up in the same commit.

erwango reviewed (changes requested):
Thanks, a quite welcome clean up. Some points to verify, though.

marwaiehm-st commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -394,37 +383,19 @@ static int eth_tx(const struct device *dev, struct net_pkt *pkt)
+	tx_config.Length = total_len;
 	tx_config.pData = &ctx;
 	tx_config.TxBuffer = &dma_tx_buffer_header[ctx.first_tx_buffer_index].tx_buff;
-#else
-	tx_config.TxBuffer = &tx_buffer_def;
-#endif /* CONFIG_ETH_STM32_HAL_API_V2 */
```
> Are you sure these lines should be removed? In non API V2 case, they will be missing and this makes the remaining part of the function API V2 specific, which doesn't look right. Or it should be added line 489.

Yes, it should be removed because it is included in the:

```c
#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32H5X) || \
	defined(CONFIG_ETH_STM32_HAL_API_V2)
``` 
so the else is not handling the case of all non-API V2, but only the case of:

```c
#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32H5X)
``` 
Now the V1 API support has been removed for STM32H7 and STM32H5, so there is no need for this.

marwaiehm-st replied:
> Note also that the code under line 452 `#ifndef CONFIG_ETH_STM32_HAL_API_V2` should now be removed.

Yes, I will remove it.

mathieuchopstm commented on drivers/ethernet/eth_stm32_hal.c:
```diff
@@ -65,7 +65,7 @@ static const struct device *eth_stm32_phy_dev = DEVICE_PHY_BY_NAME(0);
 #endif
 
-#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32H5X)
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_ethernet)
```
For readability, I think the replacement of `#if SOC_H7 || SOC_H5` with `HAS_COMPAT(h7_eth)` should be split to a separate commit.

marwaiehm-st replied:
ok, i will put it in a separate commit

mathieuchopstm reviewed (changes requested):
Small details but otherwise LGTM.

marwaiehm-st commented on dts/bindings/ethernet/st,stm32-ethernet.yaml:
```diff
@@ -1,22 +1,8 @@
-# Copyright (c) 2020, Linaro Limited
+# Copyright (c) 2024, STMicroelectronics
```
So, i will keep the original and the new one

marwaiehm-st committed:
drivers: ethernet: eth_stm32_hal: Use the compatible st_stm32h7_ethernet

Replace "CONFIG_SOC_SERIES_STM32H7X || CONFIG_SOC_SERIES_STM32H5X"
by the compatble st_stm32h7_ethernet

Signed-off-by: IBEN EL HADJ MESSAOUD Marwa <marwa.ibenelhadjmessaoud-ext@st.com>

marwaiehm-st committed:
drivers: ethernet: eth_stm32_hal: Correct indentation

Adjusted indentation to follow consistent coding standards.

Signed-off-by: IBEN EL HADJ MESSAOUD Marwa <marwa.ibenelhadjmessaoud-ext@st.com>
